From 3a8de682fe0e8b773eaf42c940f7de00b8aee192 Mon Sep 17 00:00:00 2001
From: Igor Opaniuk <igor.opaniuk@foundries.io>
Date: Fri, 21 Jul 2023 18:08:03 +0200
Subject: [PATCH 2/2] k3: set env variable device_type

Set environment variable "device_type", which stores device
sec state: gp, hs-fs, hs-se.

Changed to swicth to simplify the updates.
The same functionality is implemented very similar with the
CONFIG_DISPLAY_CPUINFO and we should use this instead.
Reworked-by: Jose Quaresma <jose.quaresma@foundries.io>

Upstream-Status: Inappropriate [lmp specific]
Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>
Signed-off-by: Jose Quaresma <jose.quaresma@foundries.io>
---
 arch/arm/mach-k3/common.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/arm/mach-k3/common.c b/arch/arm/mach-k3/common.c
index a371b002863..8405dca28c2 100644
--- a/arch/arm/mach-k3/common.c
+++ b/arch/arm/mach-k3/common.c
@@ -703,6 +703,20 @@ int misc_init_r(void)
 		env_set("secure_rprocs", "1");
 	}
 
+        enum k3_device_type type = get_device_type();
+
+        switch (type) {
+		case K3_DEVICE_TYPE_HS_SE:
+			env_set("device_type", "hs-se");
+			break;
+		case K3_DEVICE_TYPE_HS_FS:
+			env_set("device_type", "hs-fs");
+			break;
+		case K3_DEVICE_TYPE_GP:
+			env_set("device_type", "gp");
+			break;
+        }
+
 	return 0;
 }
 
-- 
2.47.0

